editors/code/src/ctx.ts

```diff
@@ -190,7 +191,7 @@ export class Ctx {
                 const command = `${this.config.discoverProjectRunner}.discoverWorkspaceCommand`;
                 log.info(`running command: ${command}`);
                 const project: JsonProject = await vscode.commands.executeCommand(command);
-                this.addToDiscoveredWorkspaces([project]);
+                this.setWorkspaces([project]);
```

Comment:
Maybe I'm not reading this correctly (it's quite possible I am) - but is the implication of this that if I am currently working in `project1/foo.rs` and then open a new `.rs` file in `project2/bar.rs`, I will lose r-a support in project1?

Reply:
you will not lose IDE functionality. the code _here_, in particular, runs only once when rust-analyzer is starting up.

I have changes on a branch that make the companion extension fully control rust-analyzer's view of `linkedProjects`. Relatedly, rust-project (over in `buck2/integrations/rust-project`) will key/control each workspace (the generated `rust-project.json`) based on the target label or the target's membership in the workspace functionality you added.

Reply:
The relationship between the rust-analyzer extension and the companion extension is a little weird. At startup, the rust-analyzer extension will pull data from the companion extension, but in the steady state, the companion extension will _push_ changes to the rust-analyzer extension. The code here only runs at startup.